Rocket.Chat/Managers/Model/SubscriptionManager/SubscriptionManager.swift

```
@@ -22,121 +22,16 @@ struct SubscriptionManager {
-                guard !response.isError() else { return Log.debug(response.result.string) }
-
-                currentRealm?.execute({ realm in
-                    guard let auth = AuthManager.isAuthenticated(realm: realm) else { return }
-                    auth.lastSubscriptionFetch = Date.serverDate.addingTimeInterval(-1)
-                    realm.add(auth, update: true)
-                })
+        let lastUpdate = auth.lastSubscriptionFetch
 
-                let subscriptions = List<Subscription>()
-
-                // List is used the first time user opens the app
-                let list = response.result["result"].array
-
-                // Update is used on updates
-                let updated = response.result["result"]["update"].array
-
-                currentRealm?.execute({ realm in
-                    list?.forEach { object in
-                        if let rid = object["_id"].string {
-                            if let subscription = Subscription.find(rid: rid, realm: realm) {
-                                subscription.mapRoom(object)
-                                subscriptions.append(subscription)
-                            }
-                        }
-                    }
-
-                    updated?.forEach { object in
-                        if let rid = object["_id"].string {
-                            if let subscription = Subscription.find(rid: rid, realm: realm) {
-                                subscription.mapRoom(object)
-                                subscriptions.append(subscription)
-                            }
-                        }
-                    }
-
-                    realm.add(subscriptions, update: true)
-                }, completion: {
-                    completion(response)
-                })
+        client?.fetchSubscriptions(updatedSince: lastUpdate) {
+            client?.fetchRooms(updatedSince: lastUpdate) {
+                completion?()
```

The only issue I see here is: if we've any error we're not continuing the app execution, right? Only if the error is the fallback version, of course. But what if by any reason one request fails with a 400 or 500? I think we can find a way to continue the execution here.

What do you think @cardoso?

---

I can think of a few options

1. Try again after some time (5s?)
2. Disconnect the user from the server
3. Continue execution despite outdated subscriptions/rooms

I'm torn between trying again and disconnecting the user from the server

---

@cardoso As we spoke:

1. Let's retry 3 times after 3 seconds on failure; After that, just ignore the error;
2. Always continue the flow, never logout the user;